dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1412,14 +1414,48 @@ public boolean delete(Contentlet contentlet, User user,boolean respectFrontendRo
         return true;
     }
 
-    @WrapInTransaction
+
+
     @Override
-    public boolean deleteByHost(Host host, User user, boolean respectFrontendRoles)
+    public boolean deleteByHost(final Host host, final User user, final boolean respectFrontendRoles)
             throws DotDataException, DotSecurityException {
-        List<Contentlet> contentletsToDelete = findContentletsByHost(host, user,
-                respectFrontendRoles);
 
-        return deleteContentlets(contentletsToDelete, user, respectFrontendRoles, true);
+        
+        final DotConnect db = new DotConnect();
```

Comment:
I think you can annotated the method deleteByHost in other to close if needed the connection to avoid orphan connections